app/src/components/core/metaschema/metaSchema.ts

```diff
@@ -16,102 +16,146 @@ module app.core.metaschema {
+                        // we are sure that there is a new object introduced here
+                        if (items.hasOwnProperty('properties') && items['properties'].hasOwnProperty('type')) {
+                            if (items['properties']['type'].hasOwnProperty('enum')) {
+                                _.forEach(items['properties']['type']['enum'], (name:string) => {
+                                    result.push(new Definition(name.toLowerCase(), Metaschema.cleanupDataschema(items, json), Metaschema.retrieveAcceptedElements(items, definitionsNames)));
+                                });
                             }
                         }
                     }
                 }
             }
+
             return result;
         }
 
-        private static cleanupDataschema(dataschema:{}):{} {
-            var result = {};
+        private static cleanupDataschema(dataschema:{}, json:{}):{} {
+            var result = {
+                "type": "object",
+                "properties" : {}
+            };
+            var propertiesSources = [];
+
+            if (dataschema['properties']) {
+                propertiesSources = [dataschema];
+            } else {
+                propertiesSources = dataschema['allOf'];
+            }
+            for (var i = 0; i < propertiesSources.length; i++) {
+                var propertySource = propertiesSources[i];
 
-            if (dataschema.hasOwnProperty('properties')) {
-                result['properties'] = _.omit(dataschema['properties'], ['elements']);
+                if (propertySource['properties']) {
+                    result['properties'] = _.merge(_.omit(propertySource['properties'], ['elements']), result['properties']);
 
-                if (dataschema['properties'].hasOwnProperty('scope')) {
-                    result['properties']['scope'] = {'type': 'string'};
-                }
+                    if (propertySource['properties'].hasOwnProperty('scope')) {
+                        result['properties']['scope'] = {'type': 'string'};
+                    }
 
-                if (dataschema['properties'].hasOwnProperty('type')) {
-                    if (dataschema['properties']['type'].hasOwnProperty('enum')) {
-                        if (dataschema['properties']['type']['enum'].length > 1) {
-                            result['properties']['type'] = dataschema['properties']['type'];
-                        }
-                        else {
-                            result['properties']['type'] = _.omit(dataschema['properties']['type'], ['enum']);
+                    if (propertySource['properties'].hasOwnProperty('type')) {
+                        if (propertySource['properties']['type'].hasOwnProperty('enum')) {
+                            if (propertySource['properties']['type']['enum'].length > 1) {
+                                result['properties']['type'] = propertySource['properties']['type'];
```

Comment:
As mentioned, deep nesting can be avoided by extracting a predicate that checks whether the enum value is present